Objects/dictobject.c

```diff
@@ -5667,6 +6064,19 @@ _PyObject_StoreInstanceAttribute(PyObject *obj, PyDictValues *values,
     return 0;
 }
 
+int
+_PyObject_StoreInstanceAttribute(PyObject *obj, PyDictValues *values,
+                              PyObject *name, PyObject *value)
+{
+    int res;
+    Py_BEGIN_CRITICAL_SECTION(obj);
```

Comment:
I don't think we want a critical section around the object here. We will need to do some locking higher up the call stack to handle the dict-or-values pointer.